### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [ServiceManager/ServiceManager#20]
Would you like to add more error handling for return values from functions like the following?
- [pthread_mutexattr_init](http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_mutexattr_init.html "Initialise the mutex attributes object.") ⇒ [mtx_init](https://github.com/ServiceManager/ServiceManager/blob/fc5a4955236f0605e8beed1611523429fe2fcdd8/external/c11thrd/threads.h#L123)
- [setsockopt](http://pubs.opengroup.org/onlinepubs/9699919799/functions/setsockopt.html "Set the socket options.") ⇒ [main](https://github.com/ServiceManager/ServiceManager/blob/37535a24e58ff990e375352f8e5c05cfa228612c/app/restartd/restartd.c#L66)

**Comment by elfring** [comment]
Are you interested to apply [aspect-oriented software development](http://aspectc.org/ "AspectC++")?
How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](http://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in your software?

**Comment by JX7P** [comment]
This sounds like an interesting proposal - I had never heard of aspect-oriented software development until now, but I find it appealing from the small amount I've read from your links. 

I had actually considered trying what is known as 'literate programming' in S16, and this shares the concept of 'tangling' - in literate programming, one attempts to expound on the logic of a program in an order that is appropriate to human understanding, and this 'web' is then 'tangled' into compiler-ready source.

I will investigate further AOP and I will report back here when I have understood its principles and thought on its use. Thank you for showing me this.

**Comment by elfring** [comment]
Are there [a few update candidates](https://github.com/ServiceManager/ServiceManager/blob/37535a24e58ff990e375352f8e5c05cfa228612c/lib/s16db/translate.c#L11 "Update candidate: property_to_rpc_property()") left over?

How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2FC.2B.2B "Tools for static code analysis") also for your software?

Do you find information sources like the following useful?
* Article "[A Reusable Aspect for Memory Allocation Checking](https://web.archive.org/web/20120220075303/http://research.msrg.utoronto.ca/ACC/Tutorial#A_Reusable_Aspect_for_Memory_All "AspectC Tutorial")"
* Document "[Aspect-Oriented Software Development - An Introduction](https://web.archive.org/web/20130512025224/http://www.info.ucl.ac.be/~jbrichau/courses/introductionToAOSD.pdf "AOSD")" by Johan Brichau and Theo D'Hondt
* Paper "[On Typesafe Aspect Implementations in C++](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.125.2164 "AspectC++")" by Daniel Lohmann and Olaf Spinczyk

**Comment by JX7P** [comment]
I have indeed found the links interesting, thank you. In translate.c, I have since added sufficient checking for malloc failure - TI-RPC handles any null pointers that it should find there.

I have been using cppcheck - I will try to integrate clang-analyze, but it seems to have changed its use conventions away from implementing a standard Unix `cc` over to something different, and I haven't learnt how to work with that yet.

On AOP again: I have been having some trouble with building working forms of AspeCt C and aspectc++; I will comment again when I have these working and have experimented with these.
